File: capy/src/e2e-cfun/java/dev/capylang/test/ConstructorResultAssertTest.java
package dev.capylang.test;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class ConstructorResultAssertTest {
    @Test
    void constructorExpressionUsesResultAssertForSuccessChecks() {
        assertThat(ConstructorResultAssert.validDateAssertSucceeds()).isTrue();
    }

    @Test
    void constructorExpressionUsesResultAssertForFailureChecks() {
        assertThat(ConstructorResultAssert.invalidDateAssertFails()).isTrue();
    }
}

File: capy/src/e2e-cfun/js/dev/capylang/test/ConstructorResultAssert.test.js
const test = require('node:test');
const { assert, generatedModule } = require('./_support');

test('ConstructorResultAssert', () => {
    const constructorAssert = generatedModule('dev/capylang/test/ConstructorResultAssert.js');
    assert.equal(constructorAssert.validDateAssertSucceeds(), true);
    assert.equal(constructorAssert.invalidDateAssertFails(), true);
});
